configure.py

```diff
@@ -1110,46 +1110,127 @@ def set_tf_nccl_install_path(environ_cp):
     if tf_nccl_version == '1':
       break  # No need to get install path, NCCL 1 is a GitHub repo.
 
-    # TODO(csigg): Look with ldconfig first if we can find the library in paths
+    # Look with ldconfig first if we can find the library in paths
     # like /usr/lib/x86_64-linux-gnu and the header file in the corresponding
     # include directory. This is where the NCCL .deb packages install them.
-    # Then ask the user if we should use that. Instead of a single
-    # NCCL_INSTALL_PATH, pass separate NCCL_LIB_PATH and NCCL_HDR_PATH to
-    # nccl_configure.bzl
-    default_nccl_path = environ_cp.get('CUDA_TOOLKIT_PATH')
-    ask_nccl_path = (r'Please specify the location where NCCL %s library is '
+
+    # First check to see if NCCL is in the ldconfig.
+    # If its found, use that location.
+    if is_linux():
+      ldconfig_bin = which('ldconfig') or '/sbin/ldconfig'
+      nccl2_path_from_ldconfig = run_shell([ldconfig_bin, '-p'])
+      nccl2_path_from_ldconfig = re.search('.*libnccl.so .* => (.*)',
+                                           nccl2_path_from_ldconfig)
+    if nccl2_path_from_ldconfig:
+      nccl2_path_from_ldconfig = nccl2_path_from_ldconfig.group(1)
+      if os.path.exists('%s.%s' % (nccl2_path_from_ldconfig, tf_nccl_version)):
+        nccl_install_path = os.path.dirname(nccl2_path_from_ldconfig)
+        print('NCCL libraries found in ' + nccl2_path_from_ldconfig)
+        
+        # Check if this is the main Linux system lib location
+        # If so, mark the install root path as simply /usr
+        # so nccl_configure.bzl can find both the header and lib
+        if nccl_install_path.startswith("/usr/lib") and re.search('.*linux-gnu', nccl_install_path):
+          nccl_root_path = "/usr"
+          print("This looks like a system path. Using: " + nccl_root_path)
+          # Look for license
```

Comment:
I took out the part looking for the license due to the dynamic link and we do not include the license file in the build as users are not bound to it.

Reply:
Thanks @tfboyd ! So the unviewable internal build errors are caused by the inclusion of the search and collection of the license file?

>> we do not include the license file in the build as users are not bound to it.

I'll take your word for that. :)